build-info-extractor/src/test/java/org/jfrog/build/extractor/BuildExtractorUtilsTest.java

```diff
@@ -339,4 +287,81 @@ private Properties createPropertiesEnvs() {
     }
+
+    public void failToReadEncryptedFileWithNoKey() throws Exception {
+        // Create encrypted file with properties
+        setupEncryptedFileTest(createProperties());
+        // Remove key
+        unsetEnv(BuildInfoConfigProperties.PROP_PROPS_FILE_KEY);
+        // Read properties from the encrypted file
+        Properties fileProps = filterDynamicProperties(
+                mergePropertiesWithSystemAndPropertyFile(new Properties(), getLog()),
+                BUILD_INFO_PROP_PREDICATE);
+        // Check if no properties are read
+        assertEquals(fileProps.size(), 0, "0 properties should be present, the file is encrypted, and the key is not available");
+    }
+
+    private void setupEncryptedFileTest(Properties props) throws Exception {
+        props.put(BuildInfoConfigProperties.PROP_PROPS_FILE, tempFile.toString());
+        System.setProperty(BuildInfoConfigProperties.PROP_PROPS_FILE, tempFile.toString());
+        ArtifactoryClientConfiguration client = new ArtifactoryClientConfiguration(new NullLog());
+        client.fillFromProperties(props);
+
+        try (FileOutputStream fileOutputStream = new FileOutputStream(tempFile.toFile())) {
+            EncryptionKeyPair keyPair = client.persistToEncryptedPropertiesFile(fileOutputStream);
+            setEnv(BuildInfoConfigProperties.PROP_PROPS_FILE_KEY, Base64.getEncoder().encodeToString(keyPair.getSecretKey()));
+            setEnv(BuildInfoConfigProperties.PROP_PROPS_FILE_KEY_IV, Base64.getEncoder().encodeToString(keyPair.getIv()));
+        }
+    }
+
+    public void getEnvAndSysPropertiesFromEncryptedFile() throws Exception {
+        // Put system properties
+        String kokoKey = "koko";
+        String gogoKey = "gogo";
+        System.setProperty(kokoKey, "parent");
+        System.setProperty(gogoKey, "2");
+
+        // Encrypt properties and write to the file
+        setupEncryptedFileTest(createPropertiesEnvs());
+
+        // Read properties from the encrypted file
+        Properties buildInfoProperties = getEnvProperties(new Properties(), new NullLog());
+
+        // Check if decrypted properties are as expected
+        assertEquals(buildInfoProperties.getProperty(ENV_POPO_KEY), "buildname", "popo property does not match");
+        assertEquals(buildInfoProperties.getProperty(ENV_MOMO_KEY), "1", "momo number property does not match");
+        assertEquals(buildInfoProperties.getProperty("koko"), "parent", "koko parent name property does not match");
+        assertEquals(buildInfoProperties.getProperty("gogo"), "2", "gogo parent number property does not match");
+
+        System.clearProperty(kokoKey);
+        System.clearProperty(gogoKey);
+    }
+
+    private void setEnv(String key, String value) throws Exception {
+        modifyEnv(key, value);
+    }
+
+    private void unsetEnv(String key) throws Exception {
+        modifyEnv(key, "");
+    }
+
+    // Method to modify (set/unset) environment variables using reflection
```

Comment:
Go style method comment
```suggestion
    /**
     * Modifies (set/unset) environment variables using reflection
     */
```